From b7ec0e56a13e9e4a53e4382d98f0699831e9fe98 Mon Sep 17 00:00:00 2001
Message-Id: <b7ec0e56a13e9e4a53e4382d98f0699831e9fe98.1350493760.git.minovotn@redhat.com>
In-Reply-To: <c93188b6c6973932d2adaa52e6a4920db13b4e62.1350493760.git.minovotn@redhat.com>
References: <c93188b6c6973932d2adaa52e6a4920db13b4e62.1350493760.git.minovotn@redhat.com>
From: Jeffrey Cody <jcody@redhat.com>
Date: Wed, 17 Oct 2012 05:59:33 +0200
Subject: [PATCH 20/35] block: qed image file reopen

RH-Author: Jeffrey Cody <jcody@redhat.com>
Message-id: <c214ac7ca7df1d4d213e602232254c09ef363d39.1350447475.git.jcody@redhat.com>
Patchwork-id: 43278
O-Subject: [RHEL6.4 qemu-kvm PATCH v4 20/35] block: qed image file reopen
Bugzilla: 767233
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Eric Blake <eblake@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

These are the stubs for the file reopen drivers for the qed format.

There is currently nothing that needs to be done by the qed driver
in reopen.

Signed-off-by: Jeff Cody <jcody@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit f9cb20f167ff205e37a895ee6a03d5a183ef8acf)
---
 block/qed.c | 9 +++++++++
 1 file changed, 9 insertions(+)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 block/qed.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/block/qed.c b/block/qed.c
index c111a7a..3d372a9 100644
--- a/block/qed.c
+++ b/block/qed.c
@@ -506,6 +506,14 @@ out:
     return ret;
 }
 
+/* We have nothing to do for QED reopen, stubs just return
+ * success */
+static int bdrv_qed_reopen_prepare(BDRVReopenState *state,
+                                   BlockReopenQueue *queue, Error **errp)
+{
+    return 0;
+}
+
 static void bdrv_qed_close(BlockDriverState *bs)
 {
     BDRVQEDState *s = bs->opaque;
@@ -1565,6 +1573,7 @@ static BlockDriver bdrv_qed = {
     .bdrv_probe               = bdrv_qed_probe,
     .bdrv_open                = bdrv_qed_open,
     .bdrv_close               = bdrv_qed_close,
+    .bdrv_reopen_prepare      = bdrv_qed_reopen_prepare,
     .bdrv_create              = bdrv_qed_create,
     .bdrv_co_is_allocated     = bdrv_qed_co_is_allocated,
     .bdrv_make_empty          = bdrv_qed_make_empty,
-- 
1.7.11.7

